js/modules/gomodule.go

```diff
@@ -25,10 +25,22 @@ func (gm *goModule) Instantiate(rt *sobek.Runtime) (sobek.CyclicModuleInstance,
 	if gm.exportedNames == nil {
 		named := mi.Exports().Named
 
-		gm.exportedNames = make([]string, len(named))
-		for name := range named {
-			gm.exportedNames = append(gm.exportedNames, name)
+		switch {
+		case len(named) == 0 && mi.Exports().Default != nil:
+			// If named length is 0 but default is defined, then try to work with
+			// default and extract the names of the object's properties. This
+			// behavior isn't ESM compatible, but we do want to allow defaults to
+			// be imported as namespaced object, which is also how node works.
+			if obj, ok := mi.Exports().Default.(*sobek.Object); ok {
+				gm.exportedNames = obj.GetOwnPropertyNames()
```

Comment:
I would use `ToObject` given that a user can just make `ToValue(anygoobject)` and IIRC this will not lead to a `sobek.Object` underneed on its own.

Reply:
I don't follow you.

So a user could have assigned `Default` with `ToValue(anygoobject)`, therefore the cast to `*sobek.Object` wouldn't work. So if we convert `Default` to a Sobek object first, then it would be able to work?

Reply:
ToValue is not required to return a *sobek.Object - it returns a Value. It likely does for practically all cases we have or extensions will use and I am pretty sure we just set `*sobek.Object` for most cases within k6 either way.

But still the correct way to get an *Object form Value is to use the [ToObject](https://pkg.go.dev/github.com/grafana/sobek#Value) funciton on the Value not to do direct cast.

Reply:
Ok, i understand the part about sobek `Value` 👍 

So since `Default` is an `interface{}`, I'd first need to convert it to a value with `rt.ToValue(Default)`, and then convert that to an object with `ToObject`.

I made the change in 296f8c286baa940621685b0a606536b955fc3792